lib/fluent/plugin/out_http.rb

```diff
@@ -100,11 +102,25 @@ def initialize
       @uri = nil
       @proxy_uri = nil
       @formatter = nil
+
+      @cache = []
+      @cache_id_mutex = Mutex.new
+      @cache_entry = Struct.new(:uri, :conn)
```

Comment:
I believe that we can manage these caches more simply.

```suggestion
      @cache_by_thread = {}
```

Why do you define the struct in the constructor?
It would be normal to define it directly under the class.

```ruby
class HTTPOutput < Output
  Fluent::Plugin.register_output('http', self)

  class RetryableResponse < StandardError; end

  CacheEntry = Struct.new(:uri, :conn)
  ...
```

Reply:
> Why do you define the struct in the constructor?
It would be normal to define it directly under the class.

I fixed this in d57402eb5a5e9e24cb08d1c3a5dcc50f4f8b8dcd